### PR TITLE
BUG: clip raising with na series bounds for datetimes or ea int

### LINKED ISSUE #44785: BUG: Confusing error message when clipping datetimes
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
import pandas as pd


pd.Series(
    [1.0, 1.0],
    dtype='datetime64[ns]'
).clip(
    pd.Series(
        [None, 1.0],
        dtype='datetime64[ns]')
)
```

### Issue Description

Clipping with columns that contain NaT values raises "TypeError: '>=' not supported between instances of 'Timestamp' and 'float'".

This can be quite confusing, particularly if you're starting with float data and casting to datetime. 

### Expected Behavior

Ideally the error message would read: "TypeError: '>=' not supported between instances of 'Timestamp' and 'NaT'".

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 3e1cfc528c9beb99d7ed590705aa7c937ab9592a
python           : 3.9.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.1.0
Version          : Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 1.4.0.dev0+1294.g3e1cfc528c
numpy            : 1.21.2
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 58.0.4
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.29.0
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.9.2 (dt dec pq3 ext lo64)
jinja2           : 3.0.2
IPython          : 7.28.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : 2021.11.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.5.0
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : 2021.11.0
scipy            : 1.7.3
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : 0.19.0
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #44785
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -7381,7 +7381,13 @@ def _clip_with_one_bound(self, threshold, method, axis, inplace):
         # GH 40420
         # Treat missing thresholds as no bounds, not clipping the values
         if is_list_like(threshold):
-            fill_value = np.inf if method.__name__ == "le" else -np.inf
+            method_name_le = method.__name__ == "le"
+            if is_datetime64_any_dtype(self.dtype):
+                fill_value = Timestamp.max if method_name_le else Timestamp.min
+            elif is_extension_array_dtype(self.dtype):
+                fill_value = self.max() if method_name_le else self.min()
```
Comment (phofl) [inline pandas/core/generic.py:7388]: Is there a better way to do this for ea ints?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -7381,7 +7381,13 @@ def _clip_with_one_bound(self, threshold, method, axis, inplace):
         # GH 40420
         # Treat missing thresholds as no bounds, not clipping the values
         if is_list_like(threshold):
-            fill_value = np.inf if method.__name__ == "le" else -np.inf
+            method_name_le = method.__name__ == "le"
+            if is_datetime64_any_dtype(self.dtype):
+                fill_value = Timestamp.max if method_name_le else Timestamp.min
```
Comment (jbrockmendel) [inline pandas/core/generic.py:7386]: for dt64tz shouldn't fill_value be tzaware?

Comment (phofl) [reply pandas/core/generic.py:7386]: Is it possible to create a timezone aware Timestamp ts where ts > Timestamp.max would be true?

The fill value is not show in the result, it is just a placeholder

Comment (jbrockmendel) [reply pandas/core/generic.py:7386]: > Is it possible to create a timezone aware Timestamp ts where ts > Timestamp.max would be true?

no, that comparison would always raise TypeError

Comment (phofl) [reply pandas/core/generic.py]: Hm ok, then we could use the max case for timzeones too

File: doc/source/whatsnew/v1.4.0.rst [hunk 3]
```diff
@@ -749,6 +749,7 @@ Numeric
 - Bug in :class:`DataFrame` arithmetic ops with a subclass whose :meth:`_constructor` attribute is a callable other than the subclass itself (:issue:`43201`)
 - Bug in arithmetic operations involving :class:`RangeIndex` where the result would have the incorrect ``name`` (:issue:`43962`)
 - Bug in arithmetic operations involving :class:`Series` where the result could have the incorrect ``name`` when the operands having matching NA or matching tuple names (:issue:`44459`)
+- Bug in :meth:`Series.clip` raising if bounds are a :class:`Series` with ``NA`` values for datetimes or nullable integer dtypes (:issue:`44785`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:752]: can you move to 1.5

File: pandas/core/generic.py (hunk 1 above)
Comment (jreback) [reply pandas/core/generic.py:7389]: `pandas/core/dtypes/missing/na_value_for_dtype`

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
@phofl can you rebase

**Comment by phofl** [comment]
Sorry, forgot. Will do in the coming days

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. Feel free to update and reopen when you have the time, but closing to clear out the queue

**Pull request closed without merging**